lib/components/widgets/text_field_widgets/date_input.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+
+import '../ew_text_field.dart';
+
+class DateInput extends StatefulWidget {
+  final String? hintText;
+  final String? labelText;
+  final DateTime? savedSelectedDate;
+
+  final Widget prefixIcon;
+  final Widget suffixIcon;
+  final String dateFormat;
+  final TextStyle? labelStyle;
+  final TextStyle? textStyle;
+  final StrutStyle? strutStyle;
+  final Function(DateTime)? onDatePicked;
+
+  const DateInput({
+    super.key,
+    this.hintText = 'Date of birth',
+    this.labelText = 'Date of birth',
+    this.savedSelectedDate,
+    this.prefixIcon = const Icon(Icons.cake_outlined),
+    this.suffixIcon = const Icon(Icons.calendar_today_rounded),
+    this.dateFormat = 'd MMM yyyy',
+    this.onDatePicked,
+    this.labelStyle,
+    this.textStyle,
+    this.strutStyle,
+  });
+
+  @override
+  DateInputState createState() => DateInputState();
+}
+
+class DateInputState extends State<DateInput> {
+  DateTime? selectedDate;
+
+  @override
+  initState() {
+    super.initState();
+    selectedDate = widget.savedSelectedDate;
+  }
+
+  Future<void> _selectDate(BuildContext context) async {
+    final DateTime? picked = await showDatePicker(
+      context: context,
+      initialDate: selectedDate,
+      firstDate: DateTime(1900, 1),
+      lastDate: DateTime(2100),
+    );
+    if (picked != null && picked != selectedDate) {
+      widget.onDatePicked?.call(picked);
+      setState(() {
+        selectedDate = picked;
+      });
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final DateFormat formatter = DateFormat(widget.dateFormat);
+
+    return GestureDetector(
+      onTap: () => _selectDate(context),
+      child: AbsorbPointer(
```

Comment:
review usage

Reply:
AbsorbPointer is required to show calendar modal